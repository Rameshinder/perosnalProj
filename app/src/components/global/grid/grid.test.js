import React from 'react'
import { mount } from 'enzyme'

import { mountToJson } from 'enzyme-to-json'
import 'jest-styled-components'

import {Container, BodyContainer, Column, Row, Push} from './index.js'

describe('<Grid />', () => {
  it('Component should render Container', () => {
    const wrapper = mount(<Container />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })

  it('Component should render Body Container', () => {
    const wrapper = mount(<BodyContainer />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Row', () => {
    const wrapper = mount(<Row />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Row with nogap prop', () => {
    const wrapper = mount(<Row nogap />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Row with collapse prop', () => {
    const wrapper = mount(<Row collapse />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Column', () => {
    const wrapper = mount(<Column />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Column with nogap prop', () => {
    const wrapper = mount(<Column nogap />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
  it('Component should render Push', () => {
    const wrapper = mount(<Push />)
    const tree = mountToJson(wrapper)
    expect(tree).toMatchSnapshot()
  })
})
