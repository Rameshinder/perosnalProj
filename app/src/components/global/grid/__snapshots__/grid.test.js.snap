// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> Component should render Body Container 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  height: auto;
  margin-bottom: 64px;
}

.c0::after {
  clear: both;
  display: block;
}

@media (min-width:769px) and (max-width:960px) {
  .c0 {
    width: 768px;
  }
}

@media (min-width:961px) and (max-width:1280px) {
  .c0 {
    width: 960px;
  }
}

@media (min-width:1281px) {
  .c0 {
    width: 1280px;
  }
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Column 1`] = `
.c0 {
  min-height: auto;
}

@media (min-width:1281px) {
  .c0 {
    width: calc(0);
    float: left;
  }
}

@media (min-width:961px) and (max-width:1280px) {
  .c0 {
    width: calc(0);
    float: left;
  }
}

@media (min-width:769px) and (max-width:960px) {
  .c0 {
    width: calc(0);
    float: left;
  }
}

@media (min-width:481px) and (max-width:768px) {
  .c0 {
    width: calc(0);
    float: left;
  }
}

@media (max-width:480px) {
  .c0 {
    width: calc(0);
    float: left;
  }
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Column with nogap prop 1`] = `
.c0 {
  min-height: auto;
}

@media (min-width:1281px) {
  .c0 {
    width: calc(NaN%);
    float: left;
    margin-left: 0px;
  }
}

@media (min-width:961px) and (max-width:1280px) {
  .c0 {
    width: calc(NaN%);
    float: left;
    margin-left: 0px;
  }
}

@media (min-width:769px) and (max-width:960px) {
  .c0 {
    width: calc(NaN%);
    float: left;
    margin-left: 0px;
  }
}

@media (min-width:481px) and (max-width:768px) {
  .c0 {
    width: calc(NaN%);
    float: left;
    margin-left: 0px;
  }
}

@media (max-width:480px) {
  .c0 {
    width: calc(NaN%);
    float: left;
    margin-left: 0px;
  }
}

<styled.div
  nogap={true}
>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Container 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  height: auto;
}

.c0::after {
  clear: both;
  display: block;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Push 1`] = `
.c0 {
  width: calc(0);
  float: left;
}

<styled.p>
  <p
    className="c0"
  />
</styled.p>
`;

exports[`<Grid /> Component should render Row 1`] = `
.c0 {
  display: table;
  width: calc(0);
  float: left;
}

<styled.div>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Row with collapse prop 1`] = `
.c0 {
  display: table;
}

<styled.div
  collapse={true}
>
  <div
    className="c0"
  />
</styled.div>
`;

exports[`<Grid /> Component should render Row with nogap prop 1`] = `
.c0 {
  display: table;
  width: calc(100%);
  float: left;
  margin-left: 0px;
}

<styled.div
  nogap={true}
>
  <div
    className="c0"
  />
</styled.div>
`;
